version_updates.md

```diff
@@ -0,0 +1,130 @@
+# OpenStackVersion
+
+## Overview
+
+This document describes how to use the OpenStackVersion
+CR to apply a minor updates to a deployed OpenStack
+installation (including both Controlplane and Dataplane
+Services)
+
+## Requirements
+
+ * Use of OLM (Operator Lifecycle Manager) is assumed
+ * The CSV (ClusterServiceVersion) for the openstack-operator
+   is used as the primary resource for versioning and the
+   associated ENV variables for OpenStack service containers.
+   These are typically prefixed with RELATED\_IMAGE...
+   The CSV defines the version.
+ * OpenStack operators are always upgraded first via OLM. This
+   can be either via manual or automatic. Upgrading operators
+   will trigger OpenStackVersion to provide access to a new
+   available version.
+ * And ENV called OPENSTACK\_VERSION must be set in the CSV
+   for openstack-operator.  This controls the 'availableVersion'
+   of OpenStackVersion.
+   
+
+## General Update Workflow
+
+OpenStack is installed by following the normal installation procedure. When
+an OpenStackControlplane resource is created in any given namespace an
+OpenStackVersion resource will be immediately created when the OpenStackControlPlane
+is reconciled (1 OpenStackVersion per namespace).
+
+When a new set of Operators is installed via OLM (either manually or automatically)
+the OpenStackVersion resources will reconcile and a new 'AvailableVersion' will
+be set on each CR.
+
+In order to apply an update an administrator sets the 'TargetVersion' to the
+same value as the 'AvailableVersion' on the OpenStackVersion resources. This
+triggers the OpenStackVersion to reconcile and will update the images
+on the associated OpenStackControlplane and OpenStackDataplane resources in
+that namespace.
+
+```shell
+$ oc get openstackversion
+NAME              TARGET VERSION   AVAILABLE VERSION
+openstack         1.0.0            1.0.1
+```
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+```
+
+## Custom images for Cinder Volume
+
+In order to allow administrators to configure customized Cinder volume backends
+with custom containerImages for each backend you must specify those images
+from automatic updates. This is because there is no way to fully automate
+the creation of those images at this time. To configure a custom
+Cinder volume named 'netapp' updates you can add the name 'netapp' to the
+'cinderVolumeCustomImages' map on the OpenStackVersion CR and configure
+a custom image for that volume backend. Multiple named Cinder volume backends may
+be configured in this manner.
```

Comment:
There are a couple of things that I would like to update in this text, instead of talking about custom containerImages these would be certified vendor containerImages, also the "specify those images from automatic updates" sounds weird to me, then we talk about creation of images which can be misleading...

I think a reworked paragraph could work better in the line of:

```
In order to allow administrators to configure customized Cinder volume backends
with certified 3rd party vendor containerImages for any of the backends, 
administrators will be allowed to explicitly specify those images overriding the
default containerImage during automatic updates. This is a manual operation 
because there is no way to fully automate the selection of those images at this
time. To configure a Cinder volume backend named 'pure' with a different image 
during the updates, the administrator can add the name 'netapp' to the
'cinderVolumeCustomImages' map on the OpenStackVersion CR and set a custom image
to be used for that volume backend. Multiple named Cinder volume backends may
be configured in this manner.
```

Reply:
NetApp doesn't require a custom container, so it's misleading.  I would use Pure as an example since it does need a custom container.